### ISSUE TITLE
`serviceUrl` is generating invalid URLs for `tag` search queries

### ISSUE THREAD

**Issue by lyzadanger** [hypothesis/client#2073]
The `TagList` component uses the `serviceUrl` service to generate (search) links for listed tags. These URLs are malformed, as they put quotes around the tag.

The generated URLs look like this:

`http://localhost:5000/search?q=tag:"one"`

They should look like:

`http://localhost:5000/search?q=tag%3Atwo`

These are causing HTML validation errors at present.

Part of https://github.com/hypothesis/client/issues/2059

**Comment by lyzadanger** [comment]
Looked at this a little this afternoon. Turns out the `links` service is actually returning the link with the quotes in it. Might need to chat with folks about the right way to correct this...

**Comment by lyzadanger** [comment]
There seem to be two approaches we could use to solve this:

1. Fix it on the client side by altering `serviceUrl`'s handling of this link path, `search.tag` (it can be done with ` path.replace(/"/g, '')`. Pros: surgical, no unexpected side effects, fast to implement. Cons: feels hacky.
2. Fix it by changing the returned `link` for `search.tag` (in `h`)  to not include quotation marks. Pros: The "right" way to fix this. Cons: we've been serving this link for years from our API. Changing it could break other users' clients/services. Or it may have other knock-on effects I'm not thinking of here.

**Comment by robertknight** [comment]
Slack discussion: https://hypothes-is.slack.com/archives/C1M8NH76X/p1588336303257000
